modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceUOMLocalServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.service.impl;
+
+import com.liferay.commerce.product.exception.CPInstanceUOMException;
+import com.liferay.commerce.product.exception.DuplicateCPInstanceUOMException;
+import com.liferay.commerce.product.model.CPInstanceUOM;
+import com.liferay.commerce.product.service.base.CPInstanceUOMLocalServiceBaseImpl;
+import com.liferay.portal.aop.AopService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Leo
+ */
+@Component(
+	property = "model.class.name=com.liferay.commerce.product.model.CPInstanceUOM",
+	service = AopService.class
+)
+public class CPInstanceUOMLocalServiceImpl
+	extends CPInstanceUOMLocalServiceBaseImpl {
+
+	@Override
+	public CPInstanceUOM addCPInstanceUOM(
+			long userId, BigDecimal baseDecimalQuantity, long cpInstanceId,
+			BigDecimal conversionRate, String key, Map<Locale, String> name,
+			int precision, boolean primary, double priority,
+			boolean purchasable, String sku)
+		throws PortalException {
+
+		CPInstanceUOM cpInstanceUOM = super.createCPInstanceUOM(
+			counterLocalService.increment());
+
+		_validateCPInstanceUOMBaseDecimalQuantity(baseDecimalQuantity);
+
+		_normalizeCPInstanceUOMBaseDecimalQuantity(
+			baseDecimalQuantity, precision);
+		_validateCPInstanceUOMConversionRate(conversionRate);
+		_validateCPInstanceUOMKey(cpInstanceId, 0, key);
+		_validateCPInstanceUOMSKU(sku);
+
+		User user = _userLocalService.getUser(userId);
+
+		cpInstanceUOM.setCompanyId(user.getCompanyId());
+		cpInstanceUOM.setUserId(user.getUserId());
+		cpInstanceUOM.setUserName(user.getFullName());
+
+		cpInstanceUOM.setBaseDecimalQuantity(baseDecimalQuantity);
+		cpInstanceUOM.setCPInstanceId(cpInstanceId);
+		cpInstanceUOM.setConversionRate(conversionRate);
+
+		Date date = new Date();
+
+		cpInstanceUOM.setCreateDate(date);
+		cpInstanceUOM.setModifiedDate(date);
+
+		cpInstanceUOM.setKey(key);
+		cpInstanceUOM.setNameMap(name, user.getLocale());
```

Comment:
Let's call the parameter nameMap, also there's no need to pass default locale here